src/main/java/hudson/security/AuthorizationMatrixProperty.java

```diff
@@ -300,9 +306,9 @@ public Object unmarshal(HierarchicalStreamReader reader,
 				reader.moveDown();
 				reader.getValue(); // we used to use this but not any more.
 				reader.moveUp();
-				prop = reader.peekNextChild();
+				prop = reader.peekNextChild(); // We check the next field
 			}
-			else if ("blocksInheritance".equals(prop)) {
```

Comment:
It's a FindBugs one. Otherwise FindBugs complains that the assigned value from the previous if clause is never used. Probably it should be registered as a separate functional bug, but I was not able to find related issues